smart-contract-verifier/src/solidity/solc_cli.rs

```diff
@@ -0,0 +1,493 @@
+        pub optimize_runs: Option<usize>,
+        pub libs: BTreeMap<String, String>,
+    }
+
+    fn merge_libs(libraries: Libraries) -> BTreeMap<String, String> {
+        libraries
+            .libs
+            .into_iter()
+            .map(|(_, libs)| libs)
+            .reduce(|accum, item| accum.into_iter().chain(item).collect())
+            .unwrap_or_default()
+    }
+
+    impl TryFrom<&CompilerInput> for InputArgs {
+        type Error = SolcError;
+        fn try_from(input: &CompilerInput) -> Result<Self, Self::Error> {
+            let libs = merge_libs(input.settings.libraries.clone());
+            Ok(InputArgs {
+                optimize: input.settings.optimizer.enabled.unwrap_or(false),
+                optimize_runs: input.settings.optimizer.runs,
+                libs,
+            })
+        }
+    }
+
+    impl InputArgs {
+        pub fn build_string_args(&self) -> Vec<String> {
+            let mut vec: Vec<String> = vec![
+                "--combined-json".to_string(),
+                OutputContract::keys().join(","),
+            ];
+            if self.optimize {
+                vec.push("--optimize".to_string());
+            }
+            if let Some(runs) = self.optimize_runs {
+                // Note: --optimize-runs doesn't affect bytecode without --optimize
+                vec.push("--optimize-runs".to_string());
+                vec.push(runs.to_string());
+            }
+            if !self.libs.is_empty() {
+                vec.push("--libraries".to_string());
+                let libs: Vec<String> = self
+                    .libs
+                    .iter()
+                    .map(|(name, address)| format!("{}:{}", name, address))
+                    .collect();
+                vec.push(libs.join(","))
+            }
+            vec
+        }
+    }
+
+    #[derive(Debug)]
+    pub struct InputFiles {
+        pub files_dir: TempDir,
+        pub file_names: Vec<String>,
+    }
+
+    impl TryFrom<&CompilerInput> for InputFiles {
+        type Error = SolcError;
+        fn try_from(input: &CompilerInput) -> Result<Self, Self::Error> {
+            let files_dir = tempfile::tempdir().map_err(|e| SolcError::Message(e.to_string()))?;
+            let mut file_names = Vec::new();
+            for (name, source) in input.sources.iter() {
+                let file_path = files_dir.path().join(name);
+                let mut file = std::fs::File::create(&file_path)
+                    .map_err(|e| SolcError::Message(e.to_string()))?;
+                file_names.push(file_path.to_string_lossy().to_string());
+                file.write_all(source.content.as_bytes())
+                    .map_err(|e| SolcError::Message(e.to_string()))?;
+            }
+
+            Ok(InputFiles {
+                files_dir,
+                file_names,
+            })
+        }
+    }
+
+    impl InputFiles {
+        pub fn build_string_args(&self) -> Result<&Vec<String>, SolcError> {
```

Comment:
Nit: maybe just `build`?

Reply:
`input_files.build()`  looks like "build input files" for me

Reply:
ok maybe it make sense, not very important for me